docs/DeveloperGuide.md

```diff
@@ -1,34 +1,630 @@
+* Intellij IDEA (Highly Recommended)
+
+Firstly, **fork** this repo and **clone** a copy into your computer.
+
+If you plan to use IntelliJ IDEA:
+1. **Ensure IntelliJ is configured to develop in JDK 11.**
+   1. If you are not at the welcome screen, click `File` > `Close Projects` to close any currently opened projects.
+   1. Click on `Configure` > `Structure for New Projects` > `Project Settings` > `Project`, ensure
+      the `Project SDK` is using **JDK 11**.
+1. **Import the project as a Gradle Project.**
+   1. Click on `Import Project` and locate the `build.gradle` file and select it. Click `OK`.
+   1. If prompted, choose to `Open as Project` (not `Open as File`).
+   1. Click `OK` to accept the default settings but do ensure that the selected version of `Gradle JVM`
+      matches the JDK being used for the project.
+   1. Wait for the importing process to finish (it could take a few minutes).
+1. **Verify the setup.**
+   1. After importing successfully, locate the `src/main/java/seedu.duke/Duke.java` file, right click it, 
+      and choose `Run...`. If the setup is correct, you should see the following:
+   ![Login_Page](img/LoginPageExampleOutput.png)
+
+### 2.2 Before writing code
+
+#### 2.2.1 Configuring the Coding Style
+
+If you are using IDEA, follow this guide 
+[IDEA: Configuring the code style](https://se-education.org/guides/tutorials/intellijCodeStyle.html)
+to setup IDEA's coding style to match ours.
+
+#### 2.2.2 Set up CI
+
+There is no set up required as the project comes with GitHub Actions config files, located in `.github/workflows`
+folder. When GitHub detects these files, it will run the CI for the project automatically at each push to the master
+branch or to any PR.
+
+#### 2.2.3 Learn the Design
+
+Before starting to write any code, we recommend that you have a look at Finux's overall design by reading
+[Finux's Architecture](#31-architecture) section.
+
+## 3. Design
+
+---
+
+### 3.1 Architecture
+
+![ArchitectureDiagram](img/ArchitectureDiagram.png)\
+_Figure 1: **Finux** Architecture Diagram_
+
+> 💡 The diagrams are built with [PlantUML](https://plantuml.com/).
+> The `.puml` files can be found in the [diagrams](https://github.com/AY2021S2-CS2113T-W09-1/tp/tree/master/docs/diagrams) folder and
+> the generated picture diagrams are in the [img](https://github.com/AY2021S2-CS2113T-W09-1/tp/tree/master/docs/img) folder.
+
+The _**Architecture Diagram**_ above details the high-level design of the **Finux** Application.
+Brief explanations of the components involved are given below.
+
+The `Finux` component contains only one class `Finux`, It is responsible for,
+* At launch: Initializes the components in the correct sequence and attempts to load data from file.  
+* At shut down: Shuts down the components and invokes cleanup method where necessary.
+
+The rest of the Application consists of six components.
+* `Ui`: The user interface (Ui) of the App which handles all user input and Application output.
```

Comment:
Is there a reason why both "App" and "Application" are used here? If they refer to the same thing, it might be confusing for the reader.

Reply:
Thanks for pointing this out! Will change it.